### PR TITLE
Feature/browserstack

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Tests

### Description
This PR adds support for browserstack and changes the CI to run e2e tests on browserstack

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: browserstack.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "auth": {
+        "username": "tilmanvatteroth1",
+        "access_key": "Sp3viAhcgqcRMuZC1Mn1"
+    },
+    "browsers": [
+        {
+            "browser": "chrome",
+            "os": "Windows 10",
```
Comment (ErikMichelson) [inline browserstack.json:9]: Do we also want to add Mac tests here?
(Linux is sadly not supported)

Comment (mrdrogdrog) [reply browserstack.json:9]: Cypress doesn't support safari (yet)

Comment (ErikMichelson) [reply browserstack.json:9]: Hm, too bad. I only had a quick view into the UI and saw "Windows" and "Mac". So I thought, they'd support Safari.

Comment (mrdrogdrog) [reply browserstack.json:9]: BrowserStack supports safari (afaik). But cypress doesn't.

File: cypress/integration/upload.spec.ts [hunk 2]
```diff
@@ -41,9 +41,6 @@ describe('Upload', () => {
           link: imageUrl
         }
       })
-      cy.fixture('acme.png').then(image => {
-        this.image = image
-      })
```
Comment (DerMolly) [inline cypress/integration/upload.spec.ts:46]: Any reason you removed this to copy it into the two other tests? Did it make problems with the execution of the tests?

File: src/components/markdown-renderer/replace-components/markmap/markmap-frame.tsx [hunk 3]
```diff
@@ -46,11 +46,15 @@ export const MarkmapFrame: React.FC<MarkmapFrameProps> = ({ code }) => {
...
-      markmapLoader(svg, code)
+      try {
+        const svg: SVGSVGElement = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+        svg.setAttribute('width', '100%')
+        actualContainer.querySelectorAll('svg').forEach(child => child.remove())
+        actualContainer.appendChild(svg)
+        markmapLoader(svg, code)
+      } catch(error) {
+       // console.error(error)
+      }
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/markmap/markmap-frame.tsx]: Did we actually need this or could it be removed again? Then again maybe keep it, but uncomment the `console.error`.

File: src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ export const MermaidChart: React.FC<MermaidChartProps> = ({ code }) => {
 
   const showError = useCallback((error: string) => {
     setError(error)
-    console.error(error)
+    //console.error(error)
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx]: Should this be removed? Or did you comment it out for a test?

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/components/markdown-renderer/replace-components/markmap/markmap-frame.tsx (hunk 3 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/markmap/markmap-frame.tsx]: forgot to readd it. fixed

File: src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx (hunk 4 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx]: forgot to readd it. fixed

File: cypress/integration/upload.spec.ts (hunk 2 above)
Comment (mrdrogdrog) [reply cypress/integration/upload.spec.ts:46]: It made problems and according to the cypress docs you shouldn't do that.

File: browserstack.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            "cypress-commands": "1.1.0",
+            "cypress-file-upload": "4.1.1"
+        },
+        "package_config_options": {}
+    },
+    "connection_settings": {
+        "local": true,
+        "local_identifier": null
+    },
+    "disable_usage_reporting": false
```
Comment (ErikMichelson) [inline browserstack.json:36]: Why did you explicitly enable the CLI usage tracking?

Comment (mrdrogdrog) [reply browserstack.json:36]: Copy & paste I guess

Comment (mrdrogdrog) [reply browserstack.json]: fixed

File: .github/workflows/e2e.yml [hunk 6]
```diff
@@ -41,32 +36,36 @@ jobs:
...
-          start: 'yarn serve:build'
-      - uses: actions/upload-artifact@master
-        if: always()
+          local-testing: start
+          local-identifier: random
+
+      - name: enable cypress videos
+        run: |
+          sudo apt-get install moreutils -yqq
+          jq '.video = "true"' cypress.json | sponge cypress.json)
```
Comment (ErikMichelson) [inline .github/workflows/e2e.yml:63]:
```suggestion
          jq '.video = "true"' cypress.json | sponge cypress.json
```

Comment (mrdrogdrog) [reply .github/workflows/e2e.yml]: fixed

File: cypress/support/fill.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+declare namespace Cypress {
+  interface Chainable {
+    /**
+     * Custom command to check an external Link.
+     * @example cy.get(a#extern).checkExternalLink('http://example.com')
+     */
+    fill (value: string): Chainable<Element>
```
Comment (ErikMichelson) [inline cypress/support/fill.ts:15]: The example does not match the method.

**Review by DerMolly — APPROVED** [review]

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: cypress/support/fill.ts (hunk 7 above)
Comment (mrdrogdrog) [reply cypress/support/fill.ts:15]: fixed

File: cypress/support/fill.ts [hunk 8]
```diff
@@ -10,16 +9,15 @@ declare namespace Cypress {
   interface Chainable {
     /**
      * Custom command to check an external Link.
```
Comment (ErikMichelson) [inline cypress/support/fill.ts:11]:
```suggestion
     * Custom command to fill an input field with text and trigger a change event.
```

Sorry to bother you again. I overlooked this at first.

Comment (mrdrogdrog) [reply cypress/support/fill.ts]: done

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**